test/add_short_id/processor-spec.ts

```diff
@@ -0,0 +1,106 @@
+import 'jest-extended';
+import { DataEntity, cloneDeep, AnyObject } from '@terascope/job-components';
+import { WorkerTestHarness } from 'teraslice-test-harness';
+import DataWindow from '../../asset/src/__lib/data-window';
+
+describe('add_short_id', () => {
+    let harness: WorkerTestHarness;
+    let data: AnyObject[];
+
+    beforeEach(() => {
+        data = [
+            {
+                id: 1,
+                name: 'joe'
+            },
+            {
+                id: 2,
+                name: 'moe'
+            },
+            {
+                id: 3,
+                name: 'randy'
+            }
+        ];
+    });
+
+    async function makeTest(config: AnyObject = {}) {
+        const _op = {
+            _op: 'add_short_id',
+            field: 'unique_id',
+            length: 8
+        };
+
+        const opConfig = config ? Object.assign({}, _op, config) : _op;
+
+        harness = WorkerTestHarness.testProcessor(opConfig);
+
+        await harness.initialize();
+
+        return harness;
+    }
+
+    afterEach(async () => {
+        if (harness) await harness.shutdown();
+    });
+
+    it('should generate an empty result if no input data', async () => {
+        const test = await makeTest();
+        const results = await test.runSlice([]);
+
+        expect(results).toBeArrayOfSize(0);
+    });
+
+    it('should return docs with a unique id of the specified properties', async () => {
+        const test = await makeTest();
+
+        const results = await test.runSlice(cloneDeep(data)) as DataEntity[];
+
+        const testUnique: { [key: string]: boolean } = {};
+
+        results.forEach((doc) => {
+            expect(DataEntity.isDataEntity(doc)).toBe(true);
+            expect(doc.unique_id).toBeDefined();
+            expect(doc.unique_id).toHaveLength(8);
+
+            testUnique[doc.unique_id] = true;
+        });
+
+        expect(Object.keys(testUnique)).toHaveLength(3);
```

Comment:
It's worth keeping in mind that this will fail once every bajillion runs (e.g. when there's a collision).